scripts/extract-i18n.js

```diff
@@ -0,0 +1,158 @@
+
+        if (token.type.label === 'name' && token.value === config.translator) {
+          const parenthesis = tokens[i + 1];
+          const message = tokens[i + 2];
+
+          if (!parenthesis || parenthesis.type.label !== '(') {
+            continue;
+          }
+
+          if (!message || message.type.label !== 'string') {
+            continue;
+          }
+
+          if (messages[message.value] === '') {
+            continue;
+          }
+
+          counter += 1;
+          messages[message.value] = '';
+        }
+      }
+
+      report(
+        ` - parsing ${chalk.bold(file)} .. ${chalk.bold(
+          counter
+        )} strings found!`
+      );
+    });
+  });
+
+  return messages;
+};
+
+const generateFiles = (config, messages) => {
+  report('Generating translations:');
+
+  let exports = '/* eslint-disable quote-props */\nexport default {\n';
+
+  config.languages.forEach(language => {
+    const filepath = `${process.cwd()}/${config.messagePath}/${language}.js`;
+
+    let oldMessages = {};
+
+    try {
+      oldMessages = require(filepath);
+    } catch (e) {
+      console.log(e);
+    }
+
+    const action =
+      oldMessages && oldMessages.length === 0 ? 'generated' : 'merged';
+
+    // Get a fresh copy of found strings
+    const newMessages = { ...messages };
+
+    if (language === config.sourceLanguage) {
+      Object.keys(newMessages).forEach(key => {
+        newMessages[key] = key;
+      });
+    }
+
+    // Iterate on old strings
+    Object.keys(oldMessages).forEach(key => {
+      if (typeof newMessages[key] !== 'undefined') {
+        // Translation needed & found: copy it
+        if (oldMessages[key] !== '') {
+          newMessages[key] = oldMessages[key];
+        }
+      } else if (typeof oldMessages[key] === 'object') {
+        // old syntax, copy blindly
+        newMessages[key] = oldMessages[key];
+      } else {
+        // Translation old & gone: surround it with @
+        newMessages[key] = `@${oldMessages[key].replace(/^@+|@+$/g, '')}@`;
+      }
+    });
+
+    // Write the new message file
+    fs.writeFile(
+      filepath,
+      `module.exports = ${JSON.stringify(newMessages, null, 2)};`,
+      'utf8',
+      err => {
+        if (err) throw err;
+      }
+    );
+    report(` - ${action} ${chalk.bold(filepath)}`);
+    exports += `  '${language}': require('./${language}'),\n`;
+  });
+
+  exports += '};\n';
+
+  fs.writeFile(
+    `${process.cwd()}/${config.messagePath}/index.js`,
+    exports,
+    'utf8',
+    err => {
+      if (err) throw err;
+    }
+  );
+};
+
+const extractMessages = config => {
+  report(
+    `Extracting messages in ${chalk.green(
+      config.sourcePath
+    )} using ${chalk.green(config.translator)}()`
+  );
+
+  const messages = extractFromSource(config);
+
+  report('Done parsing\n');
+
+  generateFiles(config, messages);
+  report('Done, have fun translating!\n'); // Et voilà!
+
+  return true;
+};
+
+extractMessages(i18nConfig);
```

Comment:
ABSOLUTELY AMAZING 👏 👏 

Is the idea eventually once we phase out `translate(...)` that this will work for all of of our translations since this only parses and searches for `t(...)`?